src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -1423,18 +1486,20 @@ and Transaction(client: IAmazonDynamoDB, ?metricsCollector: (RequestMetrics -> u
         if transactionItems.Count = 0 || transactionItems.Count > 100 then
             raise
             <| System.ArgumentOutOfRangeException(nameof transactionItems, "must be between 1 and 100 items.")
+        use activity = Diagnostics.startMultipleTableActivity (transactionItems |> Seq.map tableNameForItem) "TransactWriteItems" []
         let req = TransactWriteItemsRequest(ReturnConsumedCapacity = returnConsumedCapacity, TransactItems = transactionItems)
         clientRequestToken |> Option.iter (fun x -> req.ClientRequestToken <- x)
         let! ct = Async.CancellationToken
         let! response = client.TransactWriteItemsAsync(req, ct) |> Async.AwaitTaskCorrect
+        if response.HttpStatusCode <> HttpStatusCode.OK then
+            failwithf "TransactWriteItems request returned error %O" response.HttpStatusCode
```

Comment:
cant fail this early obv

Reply:
I’m not 100% sure this will ever be hit - I’ll have to dig into the SDK code and see if I can work out what it does. Even if it did return a response on a 500 or 503 I’d expect the entire response object would be invalid & we’d get an NRE trying to access `ConsumedCapacity`